mono/mini/mini.c

```diff
@@ -2914,7 +2914,12 @@ create_jit_info (MonoCompile *cfg, MonoMethod *method_to_compile)
 				 */
 				ei->try_start = (guint8*)ei->try_start - cfg->backend->monitor_enter_adjustment;
 			}
-			tblock = cfg->cil_offset_to_bb [ec->try_offset + ec->try_len];
+			if (ec->try_offset + ec->try_len < header->code_size)
+				tblock = cfg->cil_offset_to_bb [ec->try_offset + ec->try_len];
+			else
+				tblock = cfg->bb_exit;
+			if (cfg->verbose_level)
```

Comment:
how about `if (G_UNLIKELY (cfg->verbose_level >= 4))`? at least that seems to be the verbosity level used around here.